oauth2_http/java/com/google/auth/oauth2/ComputeEngineCredentials.java

```diff
@@ -124,12 +129,26 @@ private ComputeEngineCredentials(
       scopeList.removeAll(Arrays.asList("", null));
       this.scopes = ImmutableSet.<String>copyOf(scopeList);
     }
+    if (defaultScopes == null) {
+      this.defaultScopes = ImmutableSet.<String>of();
+    } else {
+      List<String> scopeList = new ArrayList<String>(defaultScopes);
+      scopeList.removeAll(Arrays.asList("", null));
+      this.defaultScopes = ImmutableSet.<String>copyOf(scopeList);
```

Comment:
Same here - we don't need to persist `defaultScopes`, as they'll only ever be used when `scopes` is empty:

```java
if (this.scopes.isEmpty() && defaultScopes != null) {
    // ...
    this.scopes = ImmutableSet.<String>copyOf(defaultScopes);
}
```

Reply:
yes updated the code